### PR TITLE
NGRAPH-1790 Add SequenceReverse op support

### PR DESCRIPTION
**Pull request by louisfeng**
## Description ##
Added sequence reverse op support to the bridge.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [X] SequenceReverse op added and passes test_operator.py unit tests.

### CODE REVIEW

File: src/ngraph/ngraph_emitter.cc [hunk 1]
```diff
@@ -1261,6 +1262,24 @@ void Emitter::CreateLayerOps() {
...
+    int seq_axis = get_default(node, "axis", 0);
+    const int batch_axis = 1;
+    NgraphNodePtr sequence_length{nullptr};
+    if (use_sequence_length) {
+      sequence_length = op_map_[node->inputs_[1]];
+    } else {
+      const ngraph::Shape& data_shape = data->get_shape();
+      sequence_length = makeConstant(data->get_element_type(),
+                                     ngraph::Shape{data_shape.at(batch_axis)},
+                                     std::to_string(data_shape.at(seq_axis)));
```
Comment (mbrookhart) [inline src/ngraph/ngraph_emitter.cc]: Would a standard op::Reverse make more sense in this case?

Comment (louisfeng) [reply src/ngraph/ngraph_emitter.cc]: Thanks, I didn't look at that op. New fix will use `ngraph::op::Revervse` by default.

**Comment by cconvey** [comment]
@louisfeng : What testing have you done with this code?

Also, have you verified that the MXnet and nGraph provide the same semantics for this op, including any edge cases?

**Comment by mbrookhart** [comment]
test_operator.py hits all of the options in the op:
```
    assert_array_equal(test_wrapper(arr, xpu, use_sequence_length=False), arr1)
    assert_array_equal(test_wrapper(arr, xpu, sequence_length=[3, 3], use_sequence_length=True), arr1)
    assert_array_equal(test_wrapper(arr, xpu, sequence_length=[2, 2], use_sequence_length=True), arr2)
    assert_array_equal(test_wrapper(arr, xpu, sequence_length=[2, 3], use_sequence_length=True), arr3)
    assert_array_equal(test_wrapper(arr_4, xpu, sequence_length=seq_len_1, use_sequence_length=True), arr_5)
```

File: src/ngraph/ngraph_emitter.cc [hunk 2]
```diff
@@ -1261,6 +1262,21 @@ void Emitter::CreateLayerOps() {
 
     return mul_op;
   };
+  ngraph_op_funcs_["SequenceReverse"] =
+      [this](const NodePtr& node) -> NgraphNodePtr {
+    auto data = op_map_[node->inputs_[0]];
+    bool use_sequence_length = get_default(node, "use_sequence_length", false);
+    int seq_axis = get_default(node, "axis", 0);
+    const int batch_axis = 1;
```
Comment (cconvey) [inline src/ngraph/ngraph_emitter.cc]: Coding style question: what's the purpose of making `batch_axis` const, but leaving the other local variables non-const?

Comment (louisfeng) [reply src/ngraph/ngraph_emitter.cc]: good point, I should make all parameters const.

**Pull request merged**